restapi/src/main/java/io/stargate/web/docsapi/models/dto/UpgradeCollectionDto.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.docsapi.models.dto;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.stargate.web.docsapi.models.CollectionUpgradeType;
+import io.swagger.annotations.ApiModelProperty;
+import javax.validation.constraints.NotNull;
+
+/** The DTO for the collection upgrade. */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class UpgradeCollectionDto {
+
+  @JsonProperty("upgradeType")
+  @NotNull(message = "`upgradeType` is required to upgrade a collection")
```

Comment:
Are these annotations actually effective? I'm just wondering... the field is `private` and hidden behind the annotated constructor argument :thinking: